src/components/Table/StatefulTable.jsx

```diff
@@ -256,7 +261,8 @@ const StatefulTable = ({ data: initialData, expandedData, ...other }) => {
         },
         pagination: {
           ...view.pagination,
-          totalItems: filteredData.length,
+          totalItems:
+            (state.data?.length || 0) === filteredData.length ? totalItems : filteredData.length,
```

Comment:
I find this line combined with the totalItems line above really difficult to follow. Is it possible to use some well named constants and restructure this logic to better explain what is going on?

Something along the lines below (I haven't tried it but you get the gist)

```
const filteredCount = filteredData.length;
const paginationCount = state.view.pagination?.totalItems;
const currentlyLoadedDataCount = state.data?.length || 0;

const totalItems = filteredCount && filteredCount !== currentlyLoadedDataCount ? 
  filteredCount  : 
  (paginationCount || currentlyLoadedDataCount);
```

Perhaps this can be simplified further